From 434e062eee90d1184ca92b818bf16c597ec5150c Mon Sep 17 00:00:00 2001
From: Prasun Maiti <pmaiti@maxlinear.com>
Date: Thu, 27 May 2021 22:59:44 +0530
Subject: [PATCH] UGW_SW-59229: Lantiq serial driver: fix unbalanced
 lock/unlock in an error case

In set_termios ops, acquiring lock before setting baud rate
but if setting fails, returns without releasing lock

So, it should be released before return

Signed-off-by: Prasun Maiti <pmaiti@maxlinear.com>
---
 drivers/tty/serial/lantiq.c | 1 +
 1 file changed, 1 insertion(+)

diff --git a/drivers/tty/serial/lantiq.c b/drivers/tty/serial/lantiq.c
index 2cfd47d37f3c..3a615ef3d2f6 100755
--- a/drivers/tty/serial/lantiq.c
+++ b/drivers/tty/serial/lantiq.c
@@ -548,6 +548,7 @@ static void lqasc_set_termios(struct uart_port *port,
 	if (baud) {
 		lqasc_fdv_and_reload_get(ltq_port, baud, &fdv, &reload);
 	} else {
+		spin_unlock_irqrestore(&ltq_port->asc_lock, flags);
 		dev_err(port->dev, "%s: baud rate is zero!\n!", __func__);
 		return;
 	}
